### PR TITLE
[REM] account: remove validation on bills ref

### PR DESCRIPTION
**Pull request by MuhamedAbdallaMuhamed**
[REM] account: remove validation on bills ref

Remove validation on bills ref and allow users to have multiple bills with same ref

Reason: improve user experience with the system as it's a pain for them in a real life senario

Task-3869240

enterprise-PR#https://github.com/odoo/enterprise/pull/62395

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1944,25 +1944,6 @@ def _check_journal_move_type(self):
...
-            duplicate_move_ids = list(set(
-                move_id
-                for move_ids in (move.ids + duplicate.ids for move, duplicate in move_to_duplicate_moves.items() if duplicate)
-                for move_id in move_ids
-            ))
-            action = self.env['ir.actions.actions']._for_xml_id('account.action_move_line_form')
-            action['domain'] = [('id', 'in', duplicate_move_ids)]
-            action['views'] = [((view_id, 'list') if view_type == 'tree' else (view_id, view_type)) for view_id, view_type in action['views']]
-            raise RedirectWarning(
-                message=_("Duplicated vendor reference detected. You probably encoded twice the same vendor bill/credit note."),
```
Comment (gawa-odoo) [inline addons/account/models/account_move.py:1961]: While you don't need to make .po changes, you should remove the occurences of this message in the .pot

**Review by gawa-odoo — COMMENTED** [review]
Hi
You still have a test that is failing since you removed the check

File: addons/account/models/account_move.py (hunk 1 above)
Comment (muab-odoo) [reply addons/account/models/account_move.py:1961]: I tried to search for it in the pot file but it wasn't there

Comment (gawa-odoo) [reply addons/account/models/account_move.py:1961]: The pot can put \n and "" that may make you miss it if you search for the whole string. Search on part of it and you find https://github.com/odoo/odoo/blob/17.0/addons/account/i18n/account.pot#L5782 :smiley:

**Comment by gawa-odoo** [comment]
Oh, and I just noticed it but shouldn't this target 16.0? The warning exists there too, so there is a duplication of information there too

**Comment by muab-odoo** [comment]
@gawa-odoo 
Actually the tag of the card is master shall I retarget master or double check with olma first between master/16.0?

**Comment by gawa-odoo** [comment]
Check with Olma I would say

**Comment by muab-odoo** [comment]
@gawa-odoo 
Olma said that we can choose 16.0 or 17.0. But if we targeted 16.0. This file is actaully doesn't exist and I need to remove test from it.
https://github.com/odoo/enterprise/pull/60803/files#diff-8eaabd1da954a4de662ed4059ec2a9c50091a5ec5d6334d46465490be6a8ccf8
So, What do you think?

**Comment by gawa-odoo** [comment]
No worries :smile: 
I would say to do this first in 17.0
We will backport it in 16.0 for the Odoo part (I don't think we can open and link a branch in a fw-port)

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -1958,25 +1958,6 @@ def _check_journal_move_type(self):
...
-                action=action,
-                button_text=_("Open list"),
```
[the remark this answers is not in the document]
Comment (muab-odoo) [reply addons/account/models/account_move.py:1977]: Yes Yes, It's my bad.
I need to improve catching those.
Thanks!

**Review by gawa-odoo — APPROVED** [review]
@robodoo r+
Can you open the PR to backport it in 16.0 while you're at it?

**Comment by muab-odoo** [comment]
@gawa-odoo 
16.0-PR#https://github.com/odoo/odoo/pull/165354

**Pull request closed without merging**